commands/template_store_list.go

```diff
@@ -0,0 +1,170 @@
+	}
+
+	client := http.DefaultClient
+	res, clientErr := client.Do(req)
+	if clientErr != nil {
+		return nil, fmt.Errorf("error while requesting template list: %s\n", clientErr.Error())
+	}
+
+	if res != nil {
+		defer res.Body.Close()
+		if res.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("unexpected status code wanted: %d got: %d\n", http.StatusOK, res.StatusCode)
+		}
+
+		body, bodyErr := ioutil.ReadAll(res.Body)
+		if bodyErr != nil {
+			return nil, fmt.Errorf("error while reading data from templates body: %s\n", bodyErr.Error())
+		}
+
+		templatesInfo := TemplatesInfo{}
+		unmarshallErr := json.Unmarshal(body, &templatesInfo)
+		if unmarshallErr != nil {
+			return nil, fmt.Errorf("error while unmarshalling into templates struct: %s\n", unmarshallErr.Error())
+		}
+		return templatesInfo, nil
+	}
+	return nil, fmt.Errorf("error empty response body from: %s\n", DefaultTemplateList)
+}
+
+func formatTemplatesOutput(templates TemplatesInfo) string {
+	if source != allTemplates && source != incubatorRepo && source != officialRepo {
+		fmt.Printf("\nThe --source flag needs to have value %s, %s or %s\n", allTemplates, incubatorRepo, officialRepo)
+		return ""
+	}
+	var buff bytes.Buffer
+	lineWriter := tabwriter.NewWriter(&buff, 0, 0, 1, ' ', 0)
+	fmt.Fprintln(lineWriter)
+	if verbose {
+		fmt.Fprintf(lineWriter, "NAME\tLANGUAGE\tPLATFORM\tSOURCE\tDESCRIPTION\n")
+	} else {
+		fmt.Fprintf(lineWriter, "NAME\tSOURCE\tDESCRIPTION\n")
+	}
+	for _, template := range templates {
+		if source != allTemplates {
+			if source == incubatorRepo {
+				if template.Source == officialRepo {
+					continue
+				}
+				if verbose {
+					fmt.Fprintf(lineWriter, "%s\t%s\t%s\t%s\t%s\n",
+						template.CliLanguage,
+						template.Language,
+						template.Platform,
+						template.Source,
+						template.Description)
+				} else {
+					fmt.Fprintf(lineWriter, "%s\t%s\t%s\n",
+						template.CliLanguage,
+						template.Source,
+						template.Description)
+				}
+			} else if source == officialRepo {
+				if template.Source == incubatorRepo {
+					continue
+				}
+				if verbose {
+					fmt.Fprintf(lineWriter, "%s\t%s\t%s\t%s\t%s\n",
+						template.CliLanguage,
+						template.Language,
+						template.Platform,
+						template.Source,
+						template.Description)
+				} else {
+					fmt.Fprintf(lineWriter, "%s\t%s\t%s\n",
+						template.CliLanguage,
+						template.Source,
+						template.Description)
+				}
+			}
+		} else {
+			if verbose {
+				fmt.Fprintf(lineWriter, "%s\t%s\t%s\t%s\t%s\n",
+					template.CliLanguage,
+					template.Language,
+					template.Platform,
+					template.Source,
+					template.Description)
+			} else {
+				fmt.Fprintf(lineWriter, "%s\t%s\t%s\n",
+					template.CliLanguage,
+					template.Source,
+					template.Description)
+			}
+		}
+	}
+	lineWriter.Flush()
+
+	return buff.String()
+}
+
+type TemplatesInfo []TemplateInfo
+
+type TemplateInfo struct {
+	CliLanguage string `json:"cli-lang"`
```

Comment:
Should be `CLILanguage` but I'm not convinced on the naming. Can we change it to something else?

Reply:
I am a straight forward guy 😄 , proposals are welcome

Reply:
Went with `TemplateName` I think it is better that way